dll/win32/comctl32/rebar.c

```diff
@@ -2775,7 +2780,13 @@ REBAR_SetBandInfoT(REBAR_INFO *infoPtr, INT iBand, const REBARBANDINFOW *lprbbi,
 	  REBAR_Layout(infoPtr);
 	  InvalidateRect(infoPtr->hwndSelf, NULL, TRUE);
     }
-
+#ifdef __REACTOS__
```

Comment:
Should this be done *after* (as you currently did), or *before* the uChanged thing done just above?
Or maybe your new REBAR_ShowBand call be done **inside** REBAR_Layout ?

Reply:
You're absolutely right. The functions REBAR_Layout(infoPtr) and InvalidateRect(...) are already called when finishing the REBAR_SHowBand() function, so just call REBAR_ShowBand().

Reply:
Not sure you should handle any changed flag by a call to REBAR_ShowBand (even if that function also calls the REBAR_Layout + InvalidateRect at its end), since it also calls every time ShowWindow.

This is why I mentioned that maybe the REBAR_ShowBand may be done _inside_ REBAR_Layout instead, of course only under some circumstances.

Irrespectively of this, as the current code is, you may have noticed the following:
REBAR_Layout is (amongst others) called by REBAR_InsertBandT, which is the handler for RB_INSERTBAND(A/W) too. That message also takes a REBARBANDINFO(A/W) structure. This structure can also specify, much like what is done in the RB_SETBANDINFO(A/W) handler you are modifying here (REBAR_SetBandInfoT), the initial style of the rebar, in particular whether it is going to be hidden. However, handling that visibility is currently absent in REBAR_InsertBandT. I think it may be necessary to add it there too.